pandas/tests/test_graphics.py

```diff
@@ -553,6 +553,18 @@ def test_ts_area_lim(self):
         self.assertEqual(xmin, line[0])
         self.assertEqual(xmax, line[-1])
 
+    def test_label(self):
+        s = Series([1, 2])
+        ax = s.plot(label='LABEL', legend=True)
+        self._check_legend_labels(ax, labels=['LABEL'])
+        self.plt.close()
+        ax = s.plot(legend=True)
+        self._check_legend_labels(ax, labels=['None'])
+        self.plt.close()
+        s.name = 'NAME'
+        ax = s.plot(legend=True)
+        self._check_legend_labels(ax, labels=['NAME'])
+
```

Comment:
Can you add here a test for the case where the series has a name, but you provide a label (so add ´s.plot(label='LABEL', legend=True)´ and then check the legend label is LABEL and not NAME)?